File: src/main/java/com/example/bitsetexperiments/jackson/BitSetSerializer.java
package com.example.bitsetexperiments.jackson;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;

import java.io.IOException;
import java.util.BitSet;

public class BitSetSerializer extends JsonSerializer<BitSet> {
    @Override
    public void serialize(BitSet value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
        gen.writeStartArray();
        for (long l : value.toLongArray()) {
            gen.writeNumber(l);
        }
        gen.writeEndArray();
    }

    @Override
    public Class<BitSet> handledType() {
        return BitSet.class;
    }
}

File: src/main/kotlin/com/example/calendarexperiments/bitset/jackson/BitSetSerializer.kt
package com.example.calendarexperiments.bitset.jackson

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.SerializerProvider
import java.io.IOException
import java.util.*

class BitSetSerializer : JsonSerializer<BitSet>() {
    @Throws(IOException::class)
    override fun serialize(value: BitSet, gen: JsonGenerator, serializers: SerializerProvider) {
        gen.writeStartArray()
        for (l in value.toLongArray()) {
            gen.writeNumber(l)
        }
        gen.writeEndArray()
    }

    override fun handledType(): Class<BitSet> {
        return BitSet::class.java
    }
}